src/components/playback/DrawManager.js

```diff
@@ -0,0 +1,211 @@
+
+      if (this._draw()) {
+        this._enqueueFrame(frame)
+      } else {
+        this.pause()
+        this.needsRedraw = true
+        this.position = 0
+      }
+    }
+
+    this._enqueueFrame(frame)
+  }
+
+  _enqueueFrame (frame) {
+    this.drawLoop = requestAnimationFrame(frame)
+  }
+
+  _cancelNextFrame () {
+    cancelAnimationFrame(this.drawLoop)
+  }
+
+  _draw () {
+    if (this.needsRedraw) {
+      this._clear()
+    }
+
+    if (!this._doneDrawing() && !this.path) {
+      this._setUpLine()
+    }
+
+    if (this._doneDrawing()) {
+      if (this.playing && this.position < this.duration) {
+        return true
+      } else {
+        return false
+      }
+    }
+
+    while (this.currentLine && this.currentLine.points.length && this.position > last(this.currentLine.points).time) {
+      while (!this._doneDrawingLine()) {
+        this._addCurrentPoint()
+        this._nextPoint()
+      }
+
+      this._redrawLine()
+      this._nextLine()
+    }
+
+    while (this.currentLine && !this._doneDrawingLine() && this._currentPointIsInPast()) {
+      this._addCurrentPoint()
+      this._redrawLine()
+      this._nextPoint()
+    }
+
+    if (this.currentLine && this._doneDrawingLine()) {
+      this._nextLine()
+    }
+
+    return true
+  }
+
+  _getParsedLine (index) {
+    if (!this.parsedLines[index]) {
+      this.parsedLines[index] = JSON.parse(this.lines[index] || 'null')
+    }
+
+    return this.parsedLines[index]
+  }
+
+  _nextLine () {
+    this.lineCursor++
+    this.pointCursor = 0
+    this._setUpLine()
+  }
+
+  _nextPoint () {
+    this.pointCursor++
+  }
+
+  _setUpLine () {
+    this.currentLine = this._getParsedLine(this.lineCursor)
+
+    if (this.currentLine) {
+      this.color = this.currentLine.color_id
+      this.thickness = this.currentLine.thickness
+      this._buildPath()
+    }
+  }
+
+  _addCurrentPoint () {
+    this.points.push(this._currentPoint())
+  }
+
+  _currentPoint () {
+    return this.currentLine.points[this.pointCursor]
+  }
+
+  _doneDrawingLine () {
+    return this.pointCursor >= this.currentLine.points.length
+  }
+
+  _currentPointIsInPast () {
+    return this.currentLine.points[this.pointCursor].time <= this.position
+  }
+
+  _doneDrawing () {
+    return this.lineCursor >= this.lines.length
+  }
+
+  _buildPath () {
+    this.points = []
+    this.path = this.svg.append('path')
+      .attr('stroke', this.colors[this.color].normal)
+      .attr('stroke-width', this.thickness)
+      .attr('stroke-linecap', 'round')
+      .attr('fill', 'none')
+  }
+
+  _redrawLine () {
+    this.path.attr('d', this.lineFunction(this.points))
+  }
+
+  _clear () {
+    this.svg.selectAll('*').remove()
+    this.needsRedraw = false
+    this.lineCursor = 0
+    this.pointCursor = 0
+    this.currentLine = null
+    this.path = null
+  }
+}
+
+EventEmitter(DrawManager.prototype)
+
+/* global d3, requestAnimationFrame, cancelAnimationFrame */
```

Comment:
These comments tell the linter "ay, these are defined in the global scope, so don't get mad at me for referencing undefined variables"